packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -35,7 +35,9 @@ export interface PressProps extends PressEvents {
    * still pressed, onPressStart will be fired again. If set to `true`, the press is canceled
    * when the pointer leaves the target and onPressStart will not be fired if the pointer returns.
    */
-  shouldCancelOnPointerExit?: boolean
+  shouldCancelOnPointerExit?: boolean,
+  /** Whether text selection should be enabled on the pressable element. */
+  allowTextSelectionOnPress?: boolean
```

Comment:
Is there a case where the user has this set to false, interacts with the page, the user or the app sets this to true and things don't perform (restore/disable) correctly?

Also, this seems look a cool feature, but what components are using it?

Reply:
So by default we want this behavior for our RSP components (aka prevent text selection on press so Link/Accordion text doesn't get highlighted on long press), this option is introduced so people can opt out. As for the case you described, it should get cleaned up here https://github.com/adobe/react-spectrum/blob/64058f0ce9c997ee388e8de73017b0853f3fe5de/packages/%40react-aria/interactions/src/usePress.ts#L641-L643

The `allowTextSelectionOnPress` in the if state will be `false` since the value was declared in the previous render before the change to `allowTextSelectionOnPress=true` and thus will clean up the user-select stuff

Reply:
Did we have a specific use case for this option so far?

Reply:
@devongovett No specific use case yet, added it here cuz of what we discussed in our original slack convo. I can go ahead and remove it for now and we can introduce it if people request it?

Reply:
brought `allowTextSelectionOnPress` back when reverting to this PR's original approach. I felt like this will allow someone to opt out of our `user-select: none` modification if they want to handle it themselves or if they have a good reason to not apply that to a pressable element. Happy to remove it if people still don't want this prop